src/main/java/uk/gov/justice/digital/service/operationaldatastore/dataaccess/OperationalDataStoreRepositoryProvider.java

```diff
@@ -0,0 +1,20 @@
+package uk.gov.justice.digital.service.operationaldatastore.dataaccess;
+
+import jakarta.inject.Singleton;
+import uk.gov.justice.digital.config.JobArguments;
+
+import javax.sql.DataSource;
+
+@Singleton
+public class OperationalDataStoreRepositoryProvider {
+
+    private final JobArguments jobArguments;
+
+    public OperationalDataStoreRepositoryProvider(JobArguments jobArguments) {
+        this.jobArguments = jobArguments;
+    }
+
+    OperationalDataStoreRepository getOperationalDataStoreRepository(DataSource dataSource) {
```

Comment:
The provider is there in order to share the connection pool and still be able to mock injected dependencies